lib/server-app.js

```diff
@@ -225,7 +225,11 @@ proto._findLayerByHandler = function(handler) {
   for (var k = this._router.stack.length - 1; k >= 0; k--) {
     if (this._router.stack[k].handle === handler ||
       // NewRelic replaces the handle and keeps it as __NR_original
-      this._router.stack[k].handle['__NR_original'] === handler
+      this._router.stack[k].handle['__NR_original'] === handler ||
```

Comment:
@mschnee could you refactor the assertion to be an variable, like

```js
var OrigHasHandler = (this._router.stack[k].handle['__appdynamicsProxyInfo__'] &&
        this._router.stack[k].handle['__appdynamicsProxyInfo__']['orig'] === handler
      )
var handlerExists = this._router.stack[k].handle === handler || this._router.stack[k].handle['__NR_original'] === handler || OrigHasHandler
if (handlerExists) {
// ...code
}
```

Reply:
+1, let's clean up this complicated condition please and split it into multiple smaller conditions with their results stored in well-named variables.

E.g.

```ts
const current = this._router.stack[k].handle;

const isOriginal = current === handler;
// NewRelic replaces the handle and keeps it as __NR_original
const isNewRelic = current['__NR_original'] === handler;
 // AppDynamics replaces the handle and keeps it as __appdynamicsProxyInfo__.orig
const isAppDynamics = // ...

if (isOriginal || isNewRelic || isAppDynamics) {
  return this._router.stack[k];
} else {
 // ...
}
```

Reply:
I've cleaned up the complicated conditions to be a bit more human readable.

Unfortunately, @jannyHou , I don't have very much context into the background of this issue :(